modules/camera/camera_macos.mm

```diff
@@ -308,10 +308,15 @@ - (void)dealloc {
 
 void CameraMacOS::update_feeds() {
 #if MAC_OS_X_VERSION_MIN_REQUIRED >= 101500
-	AVCaptureDeviceDiscoverySession *session = [AVCaptureDeviceDiscoverySession discoverySessionWithDeviceTypes:[NSArray arrayWithObjects:AVCaptureDeviceTypeExternalUnknown, AVCaptureDeviceTypeBuiltInWideAngleCamera, nil] mediaType:AVMediaTypeVideo position:AVCaptureDevicePositionUnspecified];
-	NSArray *devices = session.devices;
+	AVCaptureDeviceDiscoverySession *session;
+	if (@available(macOS 14, iOS 17, *)) {
```

Comment:
We usually prefer to keep Godot buildable on older OS versions. Ideally, it should be buildable on 10.15 (so no `available` for versions above `11.1` should exist). If it's added to suppress warnings, it's better not to do it.

Reply:
Fixed this